base/experiments/ad9910_manager.py

```diff
@@ -0,0 +1,262 @@
+
+
+class AD9910Manager:
+    """A Manager class that manages the real-time operation of a specific AD9910 profile.
+
+    It supports both RAM and single-tone profiles, and may optionally include a DRG. This sill
+    always on profile 0.
+
+    Args:
+        core: Core, the ARTIQ Core instance for time control.
+    """
+    def __init__(self, core):
+        self._drg_map = DRGMap()
+        self._ram_profile_map = RAMProfileMap(core)
+        self._cfg_map = []
+        self._cplds = []
+        # Function pointers. Workaround for the zero-length list type inference issue.
+        self._load_ram_profiles_fp = self.bypass
+        self._load_drg_fp = self.bypass
+
+    class DDSConfig:
+        """A collection of basic AD9910 configurations, that does not belong to DRG or RAM.
+
+        Args:
+            dds: AD9910, the DDS that will playback the profile.
+            drg: DRG, the digital ramp generator configuration. Defaults to None.
+            ram_profile: RAMProfile, the RAM profile. Defaults to None.
+            frequency: float, the baseline DDS frequency in the FTW register. Defaults to 0.0.
+            phase: float, the baseline DDS phase offset in the POW register. Defaults to 0.0.
+            amplitude: float, the baseline DDS amplitude scale factor in the ASF register.
+                Defaults to 1.0.
+        """
+        def __init__(self, dds,
+                     drg=None, ram_profile=None, frequency=0.0, phase=0.0, amplitude=1.0):
+            self.ftw = dds.frequency_to_ftw(frequency)
+            self.pow_ = dds.turns_to_pow(phase)
+            self.asf = dds.amplitude_to_asf(amplitude)
+
+            if ram_profile is not None:
+                self.ram_destination = ram_profile.dest
+                self.ram_enable = 1
+            else:
+                self.ram_destination = 0
+                self.ram_enable = 0
+
+            if drg is not None:
+                self.drg_destination = drg.dest
+                self.drg_enable = 1
+            else:
+                self.drg_destination = 0
+                self.drg_enable = 0
+
+            self.osk_enable = 1
+            # OSK is never needed if RAM is disabled.
+            if not self.ram_enable:
+                self.osk_enable = 0
+            # Disable OSK to avoid overriding RAM with amplitude destination
+            elif ram_profile.ram_type in [RAMType.AMP, RAMType.POLAR]:
+                self.osk_enable = 0
+            # Disable OSK to avoid overriding amplitude DRG
+            elif (drg is not None) and drg.drg_type == DRGType.AMP:
+                self.osk_enable = 0
+
+    def append(self, dds, frequency_src=0.0, phase_src=0.0, amplitude_src=1.0):
+        """Append a AD9910 profile.
+
+        An AD9910 profile may or may not use the RAM or the DRG. If RAM profile is not specified,
+        a single-tone profile will be configured instead.
+
+        The DDS profile will be always loaded to profile 0.
```

Comment:
What does DDS profile mean here?

Reply:
DDS profile should mean single-tone/RAM profile, it meant to say that "everything will be effectively configured like a DDS profile, and it will be stored into "Profile 0". I edited it to state that the method transfers the parameters to profile 0 to ease the confusion.